### ISSUE TITLE
Some inquiries about the training process to reproduce the result

### ISSUE THREAD

**Issue by Yukang-Lin** [agentica-project/deepscaler#61]
Hi, thanks for sharing your great work. I have been trying to reproduce your result, but I found though the total trend like response length, or reward is similar, but the val score trend is so different in my test every time, and your official logs or other logs in issues.

![Image](https://github.com/user-attachments/assets/969250e2-21cd-436f-9007-4294bcf0b267)

Here are my questions:
1. how can I know if the training is still going well, if my val score not longer improve, should I wait or restart from the preview checkpoints. (like my logs only reach 0.32 in stage 1 and drop rapidly, should I wait longer time)
2. how to choose the context length switching points, is there some experience like steps or other metrics.
3. if I scale up model size to 7B/14B, what aspects should be modified, like hyper-params, difficulty of the training set.
4. if I change algorithm to  REINFORCE++/PPO/REMAX, will the model performs better. 

could you give me some suggestions or insights, thanks in advance!

**Comment by michaelzhiluo** [comment]
1. You should keep training until you observe the performance start to decline. In your case, you should kill the 8k run when the perf is dropping.

2. We generally increase context length when the response length clip ratio hits 2-4%.
 
3. We are attempted 7B/14B and have gotten 63% and 76% AIME for 7B/14B (Due to some perf gain (not significant), we decided its not worth publishing). The trick here is to do some data preprocessing where the problems your model trains on, the model already partially solves (i.e. filter all problems that aren't solved at all over 8 attempts or solved completely over 8 attempts).

4. I think the algorithm doesn't really matter here. GRPO is best for wall clock time due to not needing a critic. (Even dapo is similar for wall clock time, despite fewer iterations)
